### PR TITLE
yaml configuration for booting the each nodes

### PR DESCRIPTION
**Pull request by kjeom**
## Proposed changes

- Migration from `urfave/cli.v1` to `urfave/cli/v2`
  - See https://cli.urfave.org/migrate-v1-to-v2/ for full list of changes.
  - Changes to our codebase:
    - Specify flag aliases as `Flag.Aliases []string`
    - Specify flag envs as `Flag.EnvVars []string`
    - Flags are now pointers `&cli.BoolFlag{ ... }`
    - Commands are now pointers `&cli.Command{ ... }`
    - No `ctx.GlobalString()`. There is only `ctx.String()`. Same goes for each flag types.

- Flag positioning
  - Command - Flag - Args order is now important.
    ```bash
    kcn init --datadir mydata genesjs.json   # cmd-flag-args OK
    kcn --datadir mydata init genesis.json   # flag-cmd-args OK
    kcn init genesis.json --datadir mydata   # cmd-args-flag BAD
    ```

- Load flags from a YAML file
  - `ken --conf config.yaml`
  - The YAML file contains the (key, value) pairs where the key is a flag name or alias.

- Refactor
  - Move file `cmd/utils/nodecmd/nodeflags.go` to `cmd/utils/nodeflags.go`
  - Move function `checkExclusive` to `utils.CheckExclusive`

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [x] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by nohkwak — APPROVED** [review]

File: cmd/homi/setup/flags.go [hunk 1]
```diff
@@ -29,382 +29,422 @@ var (
...
+		Value:   0,
 	}
 
 	deriveShaImplFlag = cli.IntFlag{
-		Name:  "deriveShaImpl",
-		Usage: "Implementation of DeriveSha [0:Original, 1:Simple, 2:Concat]",
-		Value: 0,
+		Name:    "deriveShaImpl",
+		Usage:   "Implementation of DeriveSha [0:Original, 1:Simple, 2:Concat]",
+		Aliases: []string{"genesis.derive-sha-implement"},
```
Comment (blukat29) [inline cmd/homi/setup/flags.go:157]:
```suggestion
		Aliases: []string{"genesis.derive-sha-impl"},
```
either `impl` or `implementation` seems appropriate.

File: cmd/utils/flags.go [hunk 2]
```diff
@@ -72,1246 +72,1472 @@ var (
...
 	// TODO-Klaytn-Bootnode: redefine networkid
 	NetworkIdFlag = cli.Uint64Flag{
-		Name:   "networkid",
-		Usage:  "Network identifier (integer, 8217=Cypress (Mainnet) , 1000=Aspen, 1001=Baobab)",
-		Value:  cn.GetDefaultConfig().NetworkId,
-		EnvVar: "KLAYTN_NETWORKID",
+		Name:    "networkid",
+		Usage:   "Network identifier (integer, 8217=Cypress (Mainnet) , 1000=Aspen, 1001=Baobab)",
+		Value:   cn.GetDefaultConfig().NetworkId,
+		Aliases: []string{},
```
Comment (blukat29) [inline cmd/utils/flags.go:127]: Is it different from `p2p.network-id` ?

File: cmd/utils/flags.go [hunk 3]
```diff
@@ -72,1246 +72,1472 @@ var (
...
+		EnvVars: []string{"KLAYTN_STATE_TRIE_CACHE_SAVE_PERIOD"},
 	}
+
 	SenderTxHashIndexingFlag = cli.BoolFlag{
-		Name:   "sendertxhashindexing",
-		Usage:  "Enables storing mapping information of senderTxHash to txHash",
-		EnvVar: "KLAYTN_SENDERTXHASHINDEXING",
+		Name:    "sendertxhashindexing",
+		Usage:   "Enables storing mapping information of senderTxHash to txHash",
+		Aliases: []string{"common.sender-tx-gash-indexing"},
```
Comment (blukat29) [inline cmd/utils/flags.go:496]:
```suggestion
		Aliases: []string{"common.sender-tx-hash-indexing"},
```
Typo?

Comment (kjeom) [reply cmd/utils/flags.go:496]: It's typo. Thanks!

File: cmd/conf/node/default.yaml [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+  port: 8551
+  api: ""
+  gascap: 0
+  eth-tx-feecap: 0
+  read-timeout: 30
+  write-timeout: 30
+  idle-timeout: 120
+  execution-timeout: 30
+  concurrency-limit: 3000
+  # cors-domain: ""
```
Comment (blukat29) [inline cmd/conf/node/default.yaml:174]: `corsdomain` or `cors-domain`?

File: cmd/conf/node/default.yaml [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+  enable: false
+  addr: localhost
+  port: 8553
+
+ipc: 
+  disable: false
+  path: ""
+
+console:
+  jspath: .
```
Comment (blukat29) [inline cmd/conf/node/default.yaml:199]: `jspath` or `js-path`?

File: cmd/utils/flags.go (hunk 2 above)
Comment (kjeom) [reply cmd/utils/flags.go:127]: Same thing. missed the alias. Thanks for your review in detail.

File: cmd/conf/node/default.yaml (hunk 4 above)
Comment (kjeom) [reply cmd/conf/node/default.yaml:174]: cors-domain, fixed it.

File: cmd/conf/node/default.yaml (hunk 5 above)
Comment (kjeom) [reply cmd/conf/node/default.yaml:199]: js-path, fixed it.

File: cmd/conf/db/migration.yaml [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+        region: ap-northeast-2
+        is-provisioned: false
+        read-capacity: 10000
+        write-capacity: 10000
+  dst: 
+    dbtype: LevelDB
+    datadir: ""
+    single: false
+    db: 
+      leveldb:
```
Comment (blukat29) [inline cmd/conf/db/migration.yaml:23]: sample is `migration.dst.db.leveldb.*` but flags.go says `igration.dst.leveldb.*`. Same for `migration.dst.db.dynamo`.

**Review by blukat29 — APPROVED** [review]
LGTM except merge conflict

**Comment by kjeom** [comment]
https://github.com/urfave/cli/pull/1669

File: cmd/utils/config_test.go [hunk 7]
```diff
@@ -0,0 +1,272 @@
...
+		"kas.sc.anchor.request.timeout":             false,
+	}
+
+	printFlags := func(ctx *cli.Context) error {
+		for _, flag := range AllNodeFlags() {
+			expected := testcases[flag.Names()[0]]
+			actual := ctx.IsSet(flag.Names()[0])
+			if expected != actual {
+				return fmt.Errorf("flag IsSet expected %t, but return %t", expected, actual)
+			}
```
Comment (yoomee1313) [inline cmd/utils/config_test.go:263]: What about using assert.Equal()?
```suggestion
			assert.Equal(t, testcases[flag.Names()[0]], ctx.IsSet(flag.Names()[0]), "IsSet returned unexpected result")
```

File: cmd/homi/setup/cmd.go [hunk 8]
```diff
@@ -1107,13 +1105,17 @@ func indexGenType(genTypeFlag string, base string) int {
 func findGenType(ctx *cli.Context) int {
 	var genType int
 	if ctx.Args().Present() {
-		genType = indexGenType(ctx.Args()[0], "")
+		genType = indexGenType(ctx.Args().First(), "")
```
Comment (yoomee1313) [inline cmd/homi/setup/cmd.go:1108]: It's quite confusing to me. What about next?
```
	var (
		genTypeName string
		baseString string
		genType int
	)

	if ctx.Args().Present() {
		genTypeName, baseString = ctx.Args().First(), ""
	} else {
		genTypeName, baseString = ctx.String(genTypeFlag.Name), Types[0]
	}

	genType = indexGenType(genTypeName, baseString)
	if genType == TypeNotDefined {
		fmt.Printf("Wrong Type : %s\nSupported Types : [docker, local, remote, deploy]\n\n", genTypeName)
		cli.ShowSubcommandHelp(ctx)
		os.Exit(1)
	}
```

File: cmd/homi/setup/cmd.go [hunk 9]
```diff
@@ -1132,14 +1134,31 @@ func removeSpacesAndLines(b []byte) string {
...
 	return nil
 }
 
+// In urfave/cli/v2 flags must come before args
+func CheckArgs(ctx *cli.Context) error {
+	args := ctx.Args().Slice()
+	for i, arg := range args {
+		if strings.HasPrefix(arg, "--") {
+			if i > 1 {
+				return fmt.Errorf("flag %s must come before arguments", arg)
```
Comment (yoomee1313) [inline cmd/homi/setup/cmd.go:1150]: Can't we just move the misplaced flags to the right of the arguments? Is there any huddle to do it?

File: cmd/conf/homi/homi.yaml [hunk 10]
```diff
@@ -10,82 +10,93 @@ output: "homi-output"
...
+deploy: 
+  rpc-port: 8551
+  ws-port: 8552
+  p2p-port: 32323
+  data-dir: /var/klay/data
+  log-dir: /var/klay/log
+  # Only for docker gen-type
+  docker: 
+    image-id: klaytn/klaytn:latest
+    fasthttp: true
```
[the remark this answers is not in the document]
Comment (kjeom) [reply cmd/conf/homi/homi.yaml:36]: How do you think about deleting the option. This yaml file should show the effective options.
When I made this file first, the fasthttp was being used.

File: cmd/homi/setup/cmd.go (hunk 9 above)
Comment (kjeom) [reply cmd/homi/setup/cmd.go:1150]: @yoomee1313 It's quite complicated since we actually get the flag data from context.
The context is in the library area.

File: cmd/conf/homi/homi.yaml [hunk 11]
```diff
@@ -10,82 +10,93 @@ output: "homi-output"
...
+      epoch: 604800
+      proposer-policy: 0
+      subgroup: 21
 
-# To enable clique consensus engine
-clique-epoch: 604800
-clique-period: 1
+    # To enable clique consensus engine
+    clique: 
+      enable: false
```
[the remark this answers is not in the document]
Comment (kjeom) [reply cmd/conf/homi/homi.yaml:87]:
```go
cliqueFlag = cli.BoolFlag{
	Name:    "clique",
	Usage:   "Use Clique consensus",
	Aliases: []string{},
}
```
I made it as clique.enable but I missed the alias setting in that flag.
Thanks

[the remark this answers is not in the document]
Comment (kjeom) [reply cmd/conf/homi/homi.yaml:87]: Anyway I think it's better to denote **only what we use** in the yaml file.
These files is not enough yet. We need to refine this configuration files together so that we can deploy officially.

File: cmd/homi/setup/cmd.go (hunk 8 above)
Comment (kjeom) [reply cmd/homi/setup/cmd.go]: Resolved at 5afbed2c5c92c74839b9b4df1027bf423f339a3d

File: cmd/utils/config_test.go (hunk 7 above)
Comment (kjeom) [reply cmd/utils/config_test.go]: 5afbed2c5c92c74839b9b4df1027bf423f339a3d

File: cmd/conf/node/default.yaml [hunk 12]
```diff
@@ -0,0 +1,261 @@
...
+
+vm: 
+  debug: false
+  log: 0
+  internaltx: false
+
+metrics-collection-reporting: 
+  enable: false
+  prometheus: false
+  prometheusport: 61001
```
[the remark this answers is not in the document]
Comment (kjeom) [reply cmd/conf/node/default.yaml]: Good, Thanks

File: cmd/conf/homi/homi.yaml [hunk 13]
```diff
@@ -10,82 +10,93 @@ output: "homi-output"
...
+  chain-id: 1000
+  service-chain-id: 1001 
+  funding-addr: ""
+ 
+  # This flags are only activated by default genesis-type
+  governance: false
+  governance-mode: "none"
+  governing-node: "0x0000000000000000000000000000000000000000"
+  unit-price: 0
+  derive-sha-implement: 0
```
[the remark this answers is not in the document]
Comment (kjeom) [reply cmd/conf/homi/homi.yaml]: Resolved at 5afbed2c5c92c74839b9b4df1027bf423f339a3d

File: cmd/conf/homi/homi.yaml [hunk 14]
```diff
@@ -10,82 +10,93 @@ output: "homi-output"
...
+      epoch: 604800
+      period: 1
 
-# Hard fork numbers, these should be ordered by ascending
-istanbul-compatible-blocknumber: 0
-london-compatible-blocknumber: 0
-eth-tx-type-compatible-blocknumber: 0
-magma-compatible-blocknumber: 0
-kore-compatible-blocknumber: 0
+  # Hard fork numbers, these should be ordered by ascending
```
[the remark this answers is not in the document]
Comment (kjeom) [reply cmd/conf/homi/homi.yaml:91]: Yes separated. but the block number should be ordered.

File: cmd/conf/node/default.yaml [hunk 15]
```diff
@@ -0,0 +1,261 @@
...
+  # datadir: 
+  overwrite-genesis: false
+  start-block-num: 0
+  # keystore: 
+  syncmode: full
+  garbage-collection-mode: full
+  sender-tx-hash-indexing: false
+  target-gas-limit: 4712388
+  block-extra-data: 
+  srvtype: fasthttp
```
[the remark this answers is not in the document]
Comment (kjeom) [reply cmd/conf/node/default.yaml:13]: What you mean that aliases?

File: cmd/conf/homi/homi.yaml (hunk 10 above)
[the remark this answers is not in the document]
Comment (yoomee1313) [reply cmd/conf/homi/homi.yaml:36]: If possible, please do so.

File: cmd/kpn/main.go [hunk 16]
```diff
@@ -45,37 +45,37 @@ func init() {
 	// Initialize the CLI app and start kpn
 	app.Action = nodecmd.RunKlaytnNode
 	app.HideVersion = true // we have a command to print the version
-	app.Copyright = "Copyright 2018-2023 The klaytn Authors"
-	app.Commands = []cli.Command{
+	app.Copyright = "Copyright 2018-2019 The klaytn Authors"
```
Comment (aidan-kwon) [inline cmd/kpn/main.go:48]: why did you change this?

Comment (kjeom) [reply cmd/kpn/main.go:48]: Oh, I missed it when resolving conflict. Thanks

**Review by ian0371 — COMMENTED** [review]
I get the following errors running `kcn --conf default.yaml`:
```
Mismatched type for flag 'cache.level'. Expected 'string' but actual is '<nil>'
Mismatched type for flag 'http-rpc.eth-tx-feecap'. Expected 'float64' but actual is 'int'
Mismatched type for flag 'p2p.node-key'. Expected 'string' but actual is '<nil>'
Mismatched type for flag 'p2p.node-key-hex'. Expected 'string' but actual is '<nil>'
Fatal: Options "nodekey" and "nodekeyhex" are mutually exclusive
```

We should fill the fields in `default.yaml`, and remove either nodekey/nodekeyhex.
Also, please remove **trailing whitespaces** in yamls.

**Comment by aidan-kwon** [comment]
We've decided to provide only `template.yaml` instead of `k*n.yaml` for this time. After implementing auto-yaml generating feature, we will introduce other yaml files for node operators

**Comment by kjeom** [comment]
There has been a bug in the urfave/cli latest version.
This bug affects the Int64 type flag always to be 0. [This PR](https://github.com/urfave/cli/pull/1799) fixes the bug and it will be released next version.

So I have commented out these Int64 typed flags and will be added again during testing period.

File: cmd/conf/node/default.yaml (hunk 15 above)
[the remark this answers is not in the document]
Comment (kjeom) [reply cmd/conf/node/default.yaml]: @yoomee1313 Please check it

**Review by ian0371 — COMMENTED** [review]
`kcn [cmd] --help` panics:

```
❯ kcn console --help
panic: interface conversion: interface {} is *cli.Command, not cli.Command

goroutine 1 [running]:
github.com/klaytn/klaytn/cmd/utils.NewHelpPrinter.func1({0x10494cc78, 0x14000010018}, {0x104238faf, 0x1d7}, {0x1048940a0?, 0x140000fcc60?})
        .../klaytn/cmd/utils/usage.go:72 +0xa60
...
```

File: cmd/homi/extra/cmd.go [hunk 17]
```diff
@@ -21,23 +21,22 @@ import (
...
 
-	"gopkg.in/urfave/cli.v1"
-
 	"github.com/klaytn/klaytn/common"
 	"github.com/naoina/toml"
+	"github.com/urfave/cli/v2"
 )
 
 var ExtraCommand = cli.Command{
 	Name:  "extra",
```
Comment (aidan-kwon) [inline cmd/homi/extra/cmd.go:30]: How about declaring it as `*cli.Command` type rather than adding `&`  for all usage? Not only for ExtraCommand, the command can be applied to all Command declarations.

Comment (aidan-kwon) [reply cmd/homi/extra/cmd.go:30]: Also for Flag declarations

File: cmd/conf/homi/homi.yaml [hunk 18]
```diff
@@ -10,83 +10,92 @@ output: "homi-output"
...
-magma-compatible-blocknumber: 0
-kore-compatible-blocknumber: 0
-shanghai-compatible-blocknumber: 0
+  # Hard fork numbers, these should be ordered by ascending
+  hardfork: 
+    istanbul-compatible-blocknumber: 0
+    london-compatible-blocknumber: 0
+    eth-tx-type-compatible-blocknumber: 0
+    magma-compatible-blocknumber: 0
+    kore-compatible-blocknumber: 0
```
Comment (ian0371) [inline cmd/conf/homi/homi.yaml:96]:
```suggestion
    kore-compatible-blocknumber: 0
    shanghai-compatible-blocknumber: 0
```

File: cmd/homi/setup/flags.go [hunk 19]
```diff
@@ -320,121 +345,137 @@ var (
...
-		Name:  "kore-compatible-blocknumber",
-		Usage: "koreCompatible blockNumber",
-		Value: 0,
+		Name:    "kore-compatible-blocknumber",
+		Usage:   "koreCompatible blockNumber",
+		Aliases: []string{"genesis.hardfork.kore-compatible-blocknumber"},
+		Value:   0,
 	}
 
 	kip103CompatibleBlockNumberFlag = cli.Int64Flag{
```
Comment (ian0371) [inline cmd/homi/setup/flags.go:481]: I think aliases should be added for belows

File: cmd/homi/setup/cmd.go [hunk 20]
```diff
@@ -597,8 +597,6 @@ func Gen(ctx *cli.Context) error {
 			serviceChainTest = true
 		case "clique":
 			clique = true
-		default:
-			fmt.Printf("Unknown genesis type is %s.\n", genesisType)
```
Comment (aidan-kwon) [inline cmd/homi/setup/cmd.go:601]: why is this deleted?

Comment (kjeom) [reply cmd/homi/setup/cmd.go]: Added default case again

File: cmd/conf/homi/homi.yaml (hunk 18 above)
Comment (kjeom) [reply cmd/conf/homi/homi.yaml:96]: resolved 770333a

File: cmd/homi/setup/flags.go (hunk 19 above)
Comment (kjeom) [reply cmd/homi/setup/flags.go]: It doesn't matter, anyway modified for consistency

File: cmd/utils/config.go [hunk 21]
```diff
@@ -341,126 +342,126 @@ func (kCfg *KlayConfig) SetNodeConfig(ctx *cli.Context) {
...
+	cfg.ChainDataDir = ctx.String(ChainDataDirFlag.Name)
 
-	if ctx.GlobalIsSet(KeyStoreDirFlag.Name) {
-		cfg.KeyStoreDir = ctx.GlobalString(KeyStoreDirFlag.Name)
+	if ctx.IsSet(KeyStoreDirFlag.Name) {
+		cfg.KeyStoreDir = ctx.String(KeyStoreDirFlag.Name)
 	}
-	if ctx.GlobalIsSet(LightKDFFlag.Name) {
-		cfg.UseLightweightKDF = ctx.GlobalBool(LightKDFFlag.Name)
+	if ctx.Bool(LightKDFFlag.Name) {
```
Comment (blukat29) [inline cmd/utils/config.go:356]:
```suggestion
	if ctx.IsSet(LightKDFFlag.Name) {
```

File: cmd/utils/config.go [hunk 22]
```diff
@@ -341,126 +342,126 @@ func (kCfg *KlayConfig) SetNodeConfig(ctx *cli.Context) {
...
+		cfg.UseLightweightKDF = ctx.Bool(LightKDFFlag.Name)
 	}
-	if ctx.GlobalIsSet(RPCNonEthCompatibleFlag.Name) {
-		rpc.NonEthCompatible = ctx.GlobalBool(RPCNonEthCompatibleFlag.Name)
+	if ctx.Bool(RPCNonEthCompatibleFlag.Name) {
```
Comment (blukat29) [inline cmd/utils/config.go:359]:
```suggestion
	if ctx.IsSet(RPCNonEthCompatibleFlag.Name) {
```

File: cmd/utils/config.go [hunk 23]
```diff
@@ -490,172 +491,172 @@ func (kCfg *KlayConfig) SetKlayConfig(ctx *cli.Context, stack *node.Node) {
...
+	cfg.OverwriteGenesis = ctx.Bool(OverwriteGenesisFlag.Name)
+	cfg.StartBlockNumber = ctx.Uint64(StartBlockNumberFlag.Name)
 
-	cfg.LevelDBCompression = database.LevelDBCompressionType(ctx.GlobalInt(LevelDBCompressionTypeFlag.Name))
-	cfg.LevelDBBufferPool = !ctx.GlobalIsSet(LevelDBNoBufferPoolFlag.Name)
-	cfg.EnableDBPerfMetrics = !ctx.GlobalIsSet(DBNoPerformanceMetricsFlag.Name)
-	cfg.LevelDBCacheSize = ctx.GlobalInt(LevelDBCacheSizeFlag.Name)
+	cfg.LevelDBCompression = database.LevelDBCompressionType(ctx.Int(LevelDBCompressionTypeFlag.Name))
+	cfg.LevelDBBufferPool = !ctx.Bool(LevelDBNoBufferPoolFlag.Name)
+	cfg.EnableDBPerfMetrics = !ctx.Bool(DBNoPerformanceMetricsFlag.Name)
```
Comment (blukat29) [inline cmd/utils/config.go:522]:
```suggestion
	cfg.LevelDBBufferPool = !ctx.IsSet(LevelDBNoBufferPoolFlag.Name)
	cfg.EnableDBPerfMetrics = !ctx.IsSet(DBNoPerformanceMetricsFlag.Name)
```

File: cmd/utils/config.go [hunk 24]
```diff
@@ -490,172 +491,172 @@ func (kCfg *KlayConfig) SetKlayConfig(ctx *cli.Context, stack *node.Node) {
...
-	if ctx.GlobalIsSet(ExtraDataFlag.Name) {
-		cfg.ExtraData = []byte(ctx.GlobalString(ExtraDataFlag.Name))
+	if ctx.IsSet(ExtraDataFlag.Name) {
+		cfg.ExtraData = []byte(ctx.String(ExtraDataFlag.Name))
 	}
 
-	cfg.SenderTxHashIndexing = ctx.GlobalIsSet(SenderTxHashIndexingFlag.Name)
-	cfg.ParallelDBWrite = !ctx.GlobalIsSet(NoParallelDBWriteFlag.Name)
+	cfg.SenderTxHashIndexing = ctx.Bool(SenderTxHashIndexingFlag.Name)
+	cfg.ParallelDBWrite = !ctx.Bool(NoParallelDBWriteFlag.Name)
```
Comment (blukat29) [inline cmd/utils/config.go:588]:
```suggestion
	cfg.SenderTxHashIndexing = ctx.IsSet(SenderTxHashIndexingFlag.Name)
	cfg.ParallelDBWrite = !ctx.IsSet(NoParallelDBWriteFlag.Name)
```

File: cmd/utils/config.go [hunk 25]
```diff
@@ -490,172 +491,172 @@ func (kCfg *KlayConfig) SetKlayConfig(ctx *cli.Context, stack *node.Node) {
...
+		LocalCacheSizeMiB:         ctx.Int(TrieNodeCacheLimitFlag.Name),
+		FastCacheFileDir:          ctx.String(DataDirFlag.Name) + "/fastcache",
+		FastCacheSavePeriod:       ctx.Duration(TrieNodeCacheSavePeriodFlag.Name),
+		RedisEndpoints:            ctx.StringSlice(TrieNodeCacheRedisEndpointsFlag.Name),
+		RedisClusterEnable:        ctx.Bool(TrieNodeCacheRedisClusterFlag.Name),
+		RedisPublishBlockEnable:   ctx.Bool(TrieNodeCacheRedisPublishBlockFlag.Name),
+		RedisSubscribeBlockEnable: ctx.Bool(TrieNodeCacheRedisSubscribeBlockFlag.Name),
+	}
+
+	if ctx.Bool(VMEnableDebugFlag.Name) {
```
Comment (blukat29) [inline cmd/utils/config.go:603]:
```suggestion
	if ctx.IsSet(VMEnableDebugFlag.Name) {
```

File: cmd/utils/config.go [hunk 26]
```diff
@@ -490,172 +491,172 @@ func (kCfg *KlayConfig) SetKlayConfig(ctx *cli.Context, stack *node.Node) {
...
 	}
-	if ctx.GlobalIsSet(VMLogTargetFlag.Name) {
-		if _, err := debug.Handler.SetVMLogTarget(ctx.GlobalInt(VMLogTargetFlag.Name)); err != nil {
+	if ctx.IsSet(VMLogTargetFlag.Name) {
+		if _, err := debug.Handler.SetVMLogTarget(ctx.Int(VMLogTargetFlag.Name)); err != nil {
 			logger.Warn("Incorrect vmlog value", "err", err)
 		}
 	}
-	cfg.EnableInternalTxTracing = ctx.GlobalIsSet(VMTraceInternalTxFlag.Name)
+	cfg.EnableInternalTxTracing = ctx.Bool(VMTraceInternalTxFlag.Name)
```
Comment (blukat29) [inline cmd/utils/config.go:612]:
```suggestion
	cfg.EnableInternalTxTracing = ctx.IsSet(VMTraceInternalTxFlag.Name)
```

File: cmd/utils/config.go [hunk 27]
```diff
@@ -490,172 +491,172 @@ func (kCfg *KlayConfig) SetKlayConfig(ctx *cli.Context, stack *node.Node) {
...
-		params.BlockGenerationInterval = ctx.GlobalInt64(BlockGenerationIntervalFlag.Name)
+	if ctx.IsSet(BlockGenerationIntervalFlag.Name) {
+		params.BlockGenerationInterval = ctx.Int64(BlockGenerationIntervalFlag.Name)
 		if params.BlockGenerationInterval < 1 {
 			logger.Crit("Block generation interval should be equal or larger than 1", "interval", params.BlockGenerationInterval)
 		}
 	}
-	if ctx.GlobalIsSet(BlockGenerationTimeLimitFlag.Name) {
-		params.BlockGenerationTimeLimit = ctx.GlobalDuration(BlockGenerationTimeLimitFlag.Name)
+	if ctx.Bool(BlockGenerationTimeLimitFlag.Name) {
```
Comment (blukat29) [inline cmd/utils/config.go:635]:
```suggestion
	if ctx.IsSet(BlockGenerationTimeLimitFlag.Name) {
```

File: cmd/utils/nodecmd/defaultcmd.go [hunk 28]
```diff
@@ -251,19 +216,74 @@ func BeforeRunNode(ctx *cli.Context) error {
...
 }
+
+var migrationApplied = map[*cli.Command]struct{}{}
+
+// migrateGlobalFlags makes all global flag values available in the
+// context. This should be called as early as possible in app.Before.
+//
+// Example:
+//
+//    geth account new --keystore /tmp/mykeystore --lightkdf
```
Comment (ian0371) [inline cmd/utils/nodecmd/defaultcmd.go:243]: geth -> ken

**Comment by blukat29** [comment]
@kjeom
There are some cases where `GlobalIsSet` is replaced with `Bool`.
Can we keep it `IsSet` in this PR, and refactor like below in another PR?

```diff
- if ctx.IsSet(..) {
-     cfg.Item = ctx.Bool(..)
- }
+ cfg.Item = ctx.Bool(..)
```

```diff
- cfg.Item = ctx.IsSet(..)
+ cfg.Item = ctx.Bool(..)
```

**Comment by kjeom** [comment]
> @kjeom There are some cases where `GlobalIsSet` is replaced with `Bool`. Can we keep it `IsSet` in this PR, and refactor like below in another PR?
> 
> ```diff
> - if ctx.IsSet(..) {
> -     cfg.Item = ctx.Bool(..)
> - }
> + cfg.Item = ctx.Bool(..)
> ```
> 
> ```diff
> - cfg.Item = ctx.IsSet(..)
> + cfg.Item = ctx.Bool(..)
> ```

Sure and any issues?

**Comment by blukat29** [comment]
@kjeom Because it's difficult to review. If it's not changed I can just compare side-by-side.

File: cmd/utils/config.go (hunk 22 above)
Comment (kjeom) [reply cmd/utils/config.go]: This case makes sense.
```go
cfg.LevelDBBufferPool = !ctx.IsSet(LevelDBNoBufferPoolFlag.Name)
```
But it is difference context we intended. I left these things as it is.

File: cmd/utils/config.go (hunk 23 above)
Comment (kjeom) [reply cmd/utils/config.go:522]: Please look above comment

File: cmd/utils/config.go (hunk 24 above)
Comment (kjeom) [reply cmd/utils/config.go:588]: Please look above comment

File: cmd/utils/config.go (hunk 25 above)
Comment (kjeom) [reply cmd/utils/config.go]: resolved

File: cmd/utils/nodecmd/defaultcmd.go (hunk 28 above)
Comment (kjeom) [reply cmd/utils/nodecmd/defaultcmd.go]: Reflected.

**Comment by kjeom** [comment]
> @kjeom Because it's difficult to review. If it's not changed I can just compare side-by-side.

Ok, but I have left a comment for a case.

File: cmd/utils/config.go (hunk 22 above)
Comment (blukat29) [reply cmd/utils/config.go]: Understood. `ctx.Bool` is more accurate than `ctx.IsSet` in the case of `--item=false` syntax.

**Review by blukat29 — APPROVED** [review]

**Review by ian0371 — APPROVED** [review]

File: cmd/conf/node/default.yaml (hunk 15 above)
[the remark this answers is not in the document]
Comment (yoomee1313) [reply cmd/conf/node/default.yaml]: Please ignore this comment.

File: cmd/utils/nodeflags.go [hunk 29]
```diff
@@ -0,0 +1,528 @@
...
+	altsrc.NewInt64Flag(WSReadDeadLine),
+	altsrc.NewInt64Flag(WSWriteDeadLine),
+	altsrc.NewIntFlag(WSMaxConnections),
+	altsrc.NewBoolFlag(IPCDisabledFlag),
+	altsrc.NewPathFlag(IPCPathFlag),
+	altsrc.NewIntFlag(RPCReadTimeout),
+	altsrc.NewIntFlag(RPCWriteTimeoutFlag),
+	altsrc.NewIntFlag(RPCIdleTimeoutFlag),
+	altsrc.NewIntFlag(RPCExecutionTimeoutFlag),
+}
```
Comment (aidan-kwon) [inline cmd/utils/nodeflags.go:296]: `UnsafeDebugDisableFlag`, `RPCGlobalEVMTimeoutFlag` are missing from CommonRPCFlags

**Review by aidan-kwon — APPROVED** [review]
LGTM except for @blukat29's command and my minor comment. Let's handle this in another PR

**Review by nohkwak — APPROVED** [review]
